vhost/src/vhost_user/message.rs

```diff
@@ -688,6 +690,34 @@ impl VhostUserSingleMemoryRegion {
 unsafe impl ByteValued for VhostUserSingleMemoryRegion {}
 impl VhostUserMsgValidator for VhostUserSingleMemoryRegion {}
 
+/// Get shared memory regions configuration.
+#[repr(C)]
+#[derive(Debug, Default, Clone, Copy)]
+pub struct VhostUserShMemConfig {
+    /// Total number of shared memory regions
+    pub nregions: u32,
+    /// Padding for correct alignment
+    padding: u32,
+    /// Size of each memory region
+    pub memory_sizes: [u64; 8],
```

Comment:
the spec is a bit ambiguous and i don't see a QEMU implementation yet, but it sounds like this is supposed to be variable length in the wire protocol. you could of course choose to only support 8 in this library, but then i think you still need to adjust the code to read/write these values from the socket correctly + error or something in `VhostUserShMemConfig::new` if nregions is too big

Reply:
Nice catch, the implementation seems to obey a previous version of the spec patch. In the linked version, the max number of `memory_sizes` is `VIRTIO_MAX_SHMEM_REGIONS`, which is 256.

```
+VIRTIO Shared Memory Region configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
++-------------+---------+------------+----+--------------+
+| num regions | padding | mem size 0 | .. | mem size 255 |
++-------------+---------+------------+----+--------------+
```

I will update it.